src/builtins/impls/units.rs

```diff
@@ -24,9 +25,19 @@ use anyhow::{bail, Result};
 ///
 /// Note that 'm' and 'M' are case-sensitive, to allow distinguishing between
 /// "milli" and "mega" units respectively. Other units are case-insensitive.
+#[allow(clippy::cast_precision_loss)]
 #[tracing::instrument(name = "units.parse", err)]
-pub fn parse(x: String) -> Result<i64> {
-    bail!("not implemented");
+pub fn parse(x: String) -> Result<serde_json::Value> {
+    // we're giving back serde_json::Value because per Go OPA, `parse`
+    // returns usize _or_ float, and this allows to return multiple types.
+    let p = Config::new().with_decimal();
+    if let [init @ .., prefix] = x.as_bytes() {
+        // edge case here, when 'm' is lowercase that's mili
+        if b'm'.eq(prefix) {
```

Comment:
```suggestion
    // edge case here, when 'm' is lowercase that's mili
    if let [init @ .., b'm'] = x.as_bytes() {
```

and then use an `else` to avoid the `return`